File: hw/Rodriguez/u2/hw18/Zoo/ec_edu_espe_zoo_model/animal.py
from abc import ABC, abstractmethod
from datetime import date
from .cage import Cage

class Animal(ABC):
    def __init__(self, id, description, date_onborn, gender, number_of_bones, cage):
        self.id = id
        self.description = description
        self.date_onborn = date_onborn
        self.gender = gender
        self.number_of_bones = number_of_bones
        self.cage = cage

    def __str__(self):
        return f"Animal(id={self.id}, description={self.description}, date_onborn={self.date_onborn}, gender={self.gender}, number_of_bones={self.number_of_bones}, cage={self.cage})"

    @abstractmethod
    def feed(self, food):
        pass

    def register(self, cage):
        print(f"Register the animal ---> {self.__class__.__name__} in cage {cage}")

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_description(self):
        return self.description

    def set_description(self, description):
        self.description = description

    def get_date_onborn(self):
        return self.date_onborn

    def set_date_onborn(self, date_onborn):
        self.date_onborn = date_onborn

    def get_gender(self):
        return self.gender

    def set_gender(self, gender):
        self.gender = gender

    def get_number_of_bones(self):
        return self.number_of_bones

    def set_number_of_bones(self, number_of_bones):
        self.number_of_bones = number_of_bones

    def get_cage(self):
        return self.cage

    def set_cage(self, cage):
        self.cage = cage

File: hw/ccedeno/u2/HW18ZooInPythonJavaScript/ZooInJavaScript/ec.edu.espe.Zoo.model/Animal.js
class Animal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
      this.id = id;
      this.description = description;
      this.dateOnborn = new Date(dateOnborn);
      this.gender = gender;
      this.numberOfBones = numberOfBones;
      this.cage = cage;
      this.numberOfLegs = 0;
    }
  
    toString() {
      return `Animal{id=${this.id}, description=${this.description}, dateOnborn=${this.dateOnborn}, gender=${this.gender}, numberOfBones=${this.numberOfBones}, cage=${this.cage}}`;
    }
  
    feed(food) {
      throw new Error('Method "feed" must be implemented');
    }
  
    register(cage) {
      console.log(`Register the animal ---> ${this.constructor.name} in cage ${cage}`);
    }
  
    // Getters and Setters
    getId() {
      return this.id;
    }
  
    setId(id) {
      this.id = id;
    }
  
    getDescription() {
      return this.description;
    }
  
    setDescription(description) {
      this.description = description;
    }
  
    getDateOnborn() {
      return this.dateOnborn;
    }
  
    setDateOnborn(dateOnborn) {
      this.dateOnborn = new Date(dateOnborn);
    }
  
    getGender() {
      return this.gender;
    }
  
    setGender(gender) {
      this.gender = gender;
    }
  
    getNumberOfBones() {
      return this.numberOfBones;
    }
  
    setNumberOfBones(numberOfBones) {
      this.numberOfBones = numberOfBones;
    }
  
    getCage() {
      return this.cage;
    }
  
    setCage(cage) {
      this.cage = cage;
    }
  }
  
  module.exports = Animal;
  